File: js/asyncdemo.js
"use strict"

async function return_number_async(n) {
    console.log("in return_number_async")
    return n
}

async function silly_example() {
    console.log('start of async function')
    let fut = return_number_async(10)
    console.log('called return_number_async(10)')
    let n = await fut
    console.log(`result: ${n}`)
}

silly_example()

File: rust/asyncdemo.rs
use futures::executor::block_on;

async fn return_number_async(n: i32) -> i32 {
    println!("in return_number_async");
    return n;
}

async fn silly_example() {
    println!("start of async function");
    let fut = return_number_async(10);
    println!("called return_number_async(10)");
    let n = fut.await;
    println!("result: {0}", n);
}

fn main() {
    block_on(silly_example());
}

